webapp/app/components/lists/lists.component.js

```diff
@@ -0,0 +1,104 @@
+import React, { PureComponent, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import randomColor from 'randomcolor';
+import { FormattedMessage } from 'react-intl';
+import { withStyles } from 'material-ui/styles';
+import Card, { CardHeader, CardContent, CardActions } from 'material-ui/Card';
+import Avatar from 'material-ui/Avatar';
+import DeleteIcon from '@material-ui/icons/Delete';
+import EditIcon from '@material-ui/icons/Edit';
+import IconButton from 'material-ui/IconButton';
+import ButtonBase from 'material-ui/ButtonBase';
+
+import { ListsProgress } from '../listProgress/listProgress.component';
+import styles, { Container, CardContainer, ListItems, ListItem, ProgressContainer } from './lists.styles';
+import messages from './lists.messages';
+
+export class ListsComponent extends PureComponent {
+  static propTypes = {
+    data: PropTypes.object,
+    onListClick: PropTypes.func.isRequired,
+    classes: PropTypes.object.isRequired,
+  };
+
+  getLeftItemsCount = (listItem) => listItem.get('items').filter((item) => item.get('active')).size;
+
+  getTotalItemsCount = (listItem) => listItem.get('items').size;
+
+  getAvatarColor = (seed) => randomColor({ luminosity: 'dark', seed });
+
+  handleEditClick = () => {
+    // TODO
+  };
+
+  handleDeleteClick = () => {
+    // TODO
+  };
+
+  renderAvatar = (listItem) => (
+    <Avatar aria-label="Recipe" style={{ backgroundColor: this.getAvatarColor(listItem.get('id')) }}>
+      {listItem.get('name').charAt(0).toUpperCase()}
+    </Avatar>
+  );
+
+  renderListItems = (listItem) => (
+    <ListItems>
+      {listItem.get('items').toArray().map((item, index) => (
```

Comment:
toArray is unnecessary

Reply:
Why? Has something changed? Without `.toArray` this is immutable list instead of array and in IE this causes error.